server/tests/test_pipeline.py

```diff
@@ -285,3 +285,43 @@ def test_skip_void_parameter_from_variables_for_mongo_steps():
             }
         ]
     ) == [{"$match": {"$and": [{"$or": [{"property2": "value2"}]}]}}]
+
+    assert remove_void_conditions_from_mongo_steps(
+        [
+            {"$match": {}},
+            {"$group": {"_id": None, "_vqbPipelineInline": {"$push": "$$ROOT"}}},
+            {
+                "$lookup": {
+                    "as": "_vqbPipelineToAppend_0",
+                    "from": "slide_data-append",
+                    "pipeline": [],
+                }
+            },
+            {
+                "$project": {
+                    "_vqbPipelinesUnion": {
+                        "$concatArrays": ["$_vqbPipelineInline", "$_vqbPipelineToAppend_0"]
+                    }
+                }
+            },
+            {"$unwind": "$_vqbPipelinesUnion"},
+            {"$replaceRoot": {"newRoot": "$_vqbPipelinesUnion"}},
+            {"$project": {"_id": 0}},
+        ]
+    ) == [
+        # NOTE: the $match step is rip off because of the cleaning on empty elements,
+        # it's the responsability to the client to add that {$match: {}} as the
+        # first steps
```

Comment:
This sounds dangerous. Why would it be removed ? There's some sanitization logic in toucan-connectors already, maybe you can take inspiration from there ? https://github.com/ToucanToco/toucan-connectors/blob/master/toucan_connectors/mongo/mongo_connector.py#L76

Reply:
> Why would it be removed ?

because by definition of the function, if a given key/value is given and the value is an empty dict, there is no point keeping it ! i already fix that here on laputa : https://github.com/ToucanToco/laputa/pull/5956/files#diff-20f538e207e705508e72573b79c68ceeb8454d11d85169c6838e8353d9ba2ba5R674-R675
+ 
am not sure that that sanitizer fix this related issue of the missing `$match` key only... or am wrong ? 🤔

Reply:
https://github.com/ToucanToco/weaverbird/pull/1797/commits/76ba2b3ed072815645eddc2f6f31fcc2050c3a65 i added the sanitizer method